app/src/main/java/com/vultisig/wallet/ui/models/keygen/GeneratingKeyViewModel.kt

```diff
@@ -158,17 +158,17 @@ internal class GeneratingKeyViewModel(
                 }
             }
             // here is the keygen process is done
-            val keygenVerifier = KeygenVerifier(
-                this.serverAddress,
-                this.sessionId,
-                vault.localPartyID,
-                this.keygenCommittee, gson = gson,
-            )
             withContext(Dispatchers.IO) {
-                keygenVerifier.markLocalPartyComplete()
-                if (!keygenVerifier.checkCompletedParties()) {
-                    throw Exception("another party failed to complete the keygen process")
+                keyApi.keygenMarkLocalPartyComplete(serverAddress, sessionId, listOf(vault.localPartyID))
+                Timber.d("Local party ${vault.localPartyID} marked as complete")
+                var repeate = true
+                while (repeate){
```

Comment:
Previous implementation will stop after 60 seconds , we need the same logic here as well, otherwise this will keep going forever